packages/integrations/spa/client-persist.js

```diff
@@ -0,0 +1,21 @@
+import listen from 'micromorph/spa';
+
+export default () =>
+	listen({
+		beforeDiff(doc) {
+			for (const island of doc.querySelectorAll('astro-root')) {
+				const uid = island.getAttribute('uid');
+				const current = document.querySelector(`astro-root[uid="${uid}"]`);
+				if (current) {
+					current.dataset.persist = true;
```

Comment:
`data-persist` is a special attribute that instructs `micromorph` to leave a node in place during diffing.